templates/bridge_RedHat.j2

```diff
@@ -1,21 +1,20 @@
-
 {% if item.bootproto == 'static' %}
 DEVICE={{ item.device }}
 TYPE=Bridge
 BOOTPROTO=none
-{% if item.stp is defined %}
+{% if item.stp is defined -%}
 STP={{ item.stp }}
 {% endif %}
-{% if item.address is defined %}
+{%- if item.address is defined -%}
```

Comment:
Why are the dashes there in most of the if statements e.g. '{%-' ? Is this by mistake or is this some syntax I'm not aware of?

Reply:
I added them so the network config files wouldn't end up having trailing blank lines like this:
>DEVICE=bond0                                        
>USERCTL=no                                          
>BOOTPROTO=none                                      
>ONBOOT=True                                         
>BONDING_OPTS="mode=4 miimon=100 updelay=5000 "      
>
>BRIDGE=br-mgmt

Reference: http://jinja.pocoo.org/docs/2.9/templates/#whitespace-control
And also, if I have something like this:
>BOOTPROTO=none                                                                                                                                                                                                     
>{%- if item.stp is defined -%}                                                                                                                                                                                     
>STP={{ item.stp }}                                                                                                                                                                                                 
>{% endif %}                                                                                                                                                                                                        
>{%- if item.address is defined -%}                                                                                                                                                                                 
>IPADDR={{ item.address }}                                                                                                                                                                                          
>{% endif %} 

I'll end up like this:
> BOOTPROTO=noneIPADDR=192.168.36.45

So I have to write like this:
>BOOTPROTO=none                                                                                                                                                                                                     
>{% if item.stp is defined -%}                                                                                                                                                                                     
>STP={{ item.stp }}                                                                                                                                                                                                 
>{% endif %}                                                                                                                                                                                                        
>{%- if item.address is defined -%}                                                                                                                                                                                 
>IPADDR={{ item.address }}                                                                                                                                                                                          
>{% endif %}

Reply:
Cool, TIL.